src/coverage.ts

```diff
@@ -0,0 +1,81 @@
+import { once } from 'events';
+import { fork, execSync } from 'child_process';
+import path from 'path';
+import { existsSync, renameSync } from 'fs';
+import { removeSync, moveSync, copySync } from 'fs-extra';
+import exitHook from 'exit-hook';
+
+export function cleanUp(): void {
+  if (existsSync('./contracts-backup/')) {
+    moveSync('./contracts-backup', './contracts', { overwrite: true });
+  }
+  removeSync('./contracts-backup/');
+  removeSync('./build/contracts/');
+  removeSync('./.coverage_artifacts/');
+  removeSync('./.coverage_contracts/');
+}
+
+exitHook(cleanUp);
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+export async function runCoverage(skipFiles: string[], compileCommand: string, testCommand: string[]): Promise<void> {
+  const client = require('ganache-cli');
+  const CoverageAPI = require('solidity-coverage/api');
+  const utils = require('solidity-coverage/utils');
+
+  const api = new CoverageAPI({ client });
+
+  const config = {
+    workingDir: process.cwd(),
+    contractsDir: path.join(process.cwd(), 'contracts'),
+    logger: {
+      log: (msg: string): void => console.log(msg),
+    },
+  };
+
+  try {
+    const { tempContractsDir, tempArtifactsDir } = utils.getTempLocations(config);
+    utils.setupTempFolders(config, tempContractsDir, tempArtifactsDir);
+
+    const { targets } = utils.assembleFiles(config, skipFiles);
+    const instrumented = api.instrument(targets);
+    utils.save(instrumented, config.contractsDir, tempContractsDir);
+
+    // backup original contracts
+    copySync('./contracts/', './contracts-backup', { overwrite: true });
+    copySync('./.coverage_contracts/', './contracts/', { overwrite: true });
```

Comment:
Given the above change, I think we should be able to use `moveSync` here to avoid copying the user's contracts around. In this way we only ever rename the contracts directory which should be a lot safer.

I would also remove `overwrite: true` to avoid accidentally overwriting the user's files.

```suggestion
    moveSync('./contracts/', './contracts-backup');
    moveSync(tempContractsDir, './contracts/');
```

Also note the use of the `tempContractsDir` variable rather than the hardcoded path.

Is there something else I'm missing why moving the directory like this wouldn't work?